crates/fuel-core/src/state/rocks_db_key_iterator.rs

```diff
@@ -0,0 +1,81 @@
+use crate::database::{
+    Error as DatabaseError,
+    Result as DatabaseResult,
+};
+use anyhow::anyhow;
+
+use fuel_core_storage::iter::IterDirection;
+use rocksdb::{
+    DBAccess,
+    DBRawIteratorWithThreadMode,
+};
+
+pub enum OwnedIteratorMode {
+    Start,
+    End,
+    From(Vec<u8>, rocksdb::Direction),
+}
```

Comment:
Favouring "owning" the `IteratorMode` because our keys are relatively small, and to avoid all the lifetime annotations in places which use this to cut cognitive load.